diff --git a/lib/views/swift/SwiftHybridViewManager.js b/lib/views/swift/SwiftHybridViewManager.js
index 8452c3e974f146d4b082c52d78b04a86d0da91bf..a2a54bdeac5fe152c9a2ba220d37a5b23db980e9 100644
--- a/lib/views/swift/SwiftHybridViewManager.js
+++ b/lib/views/swift/SwiftHybridViewManager.js
@@ -39,7 +39,11 @@ ${createFileMetadataString(`${component}.mm`)}
 #import <React/RCTComponentViewFactory.h>
 #import <React/UIView+ComponentViewProtocol.h>
 #import <NitroModules/NitroDefines.hpp>
+#if __has_include(<UIKit/UIKit.h>)
 #import <UIKit/UIKit.h>
+#elif __has_include(<AppKit/AppKit.h>)
+#import <AppKit/AppKit.h>
+#endif
 
 #import "${HybridTSpecSwift}.hpp"
 #import "${getUmbrellaHeaderName()}"
@@ -82,7 +86,11 @@ using namespace ${namespace}::views;
 
   // 2. Get UIView*
   void* viewUnsafe = swiftPart.getView();
+  #if __has_include(<UIKit/UIKit.h>)
   UIView* view = (__bridge_transfer UIView*) viewUnsafe;
+  #elif __has_include(<AppKit/AppKit.h>)
+  NSView* view = (__bridge_transfer NSView*) viewUnsafe;
+  #endif
 
   // 3. Update RCTViewComponentView's [contentView]
   [self setContentView:view];
diff --git a/src/views/swift/SwiftHybridViewManager.ts b/src/views/swift/SwiftHybridViewManager.ts
index 1225a25c878320bb6b734add5502e4943eee4844..cde10badeea1104dd23fab0440a7da5b6594b249 100644
--- a/src/views/swift/SwiftHybridViewManager.ts
+++ b/src/views/swift/SwiftHybridViewManager.ts
@@ -60,7 +60,11 @@ ${createFileMetadataString(`${component}.mm`)}
 #import <React/RCTComponentViewFactory.h>
 #import <React/UIView+ComponentViewProtocol.h>
 #import <NitroModules/NitroDefines.hpp>
+#if __has_include(<UIKit/UIKit.h>)
 #import <UIKit/UIKit.h>
+#elif __has_include(<AppKit/AppKit.h>)
+#import <AppKit/AppKit.h>
+#endif
 
 #import "${HybridTSpecSwift}.hpp"
 #import "${getUmbrellaHeaderName()}"
@@ -103,7 +107,11 @@ using namespace ${namespace}::views;
 
   // 2. Get UIView*
   void* viewUnsafe = swiftPart.getView();
+  #if __has_include(<UIKit/UIKit.h>)
   UIView* view = (__bridge_transfer UIView*) viewUnsafe;
+  #elif __has_include(<AppKit/AppKit.h>)
+  NSView* view = (__bridge_transfer NSView*) viewUnsafe;
+  #endif
 
   // 3. Update RCTViewComponentView's [contentView]
   [self setContentView:view];
