diff --git a/ios/views/HybridView.swift b/ios/views/HybridView.swift
index d2d52ab5d389f4a9c3243e5d47d392f29d0ea647..8a20909f28ee0328fc528d31d950f61984ee3d6c 100644
--- a/ios/views/HybridView.swift
+++ b/ios/views/HybridView.swift
@@ -5,10 +5,15 @@
 //  Created by Marc Rousavy on 13.01.25.
 //

-#if canImport(UIKit)
+

   import Foundation
+
+  #if canImport(UIKit)
   import UIKit
+  #elseif canImport(AppKit)
+  import AppKit
+  #endif

   /// A base protocol for all Swift-based Hybrid Views.
   public protocol HybridView: HybridObject {
@@ -17,7 +22,11 @@
      *
      * This value should not change during the lifetime of this ``HybridView``.
      */
+    #if canImport(UIKit)
     var view: UIView { get }
+    #elseif canImport(AppKit)
+    var view: NSView { get }
+    #endif

     /**
      * Called right before updating props.
@@ -43,4 +52,4 @@
     }
   }

-#endif
+
diff --git a/lib/commonjs/turbomodule/NativeNitroModules.js b/lib/commonjs/turbomodule/NativeNitroModules.js
index 9650b7da7d3fe76b68e2b182b28dcbf634b7d875..6f709f4fb7428b6200106b8dbee27577596907d5 100644
--- a/lib/commonjs/turbomodule/NativeNitroModules.js
+++ b/lib/commonjs/turbomodule/NativeNitroModules.js
@@ -7,10 +7,11 @@ exports.NitroModules = void 0;
 exports.isRuntimeAlive = isRuntimeAlive;
 var _reactNative = require("react-native");
 var _ModuleNotFoundError = require("../ModuleNotFoundError");
+var isTurboModuleEnabled = global.__turboModuleProxy != null;
 // 1. Get (and initialize) the TurboModule
 let turboModule;
 try {
-  turboModule = _reactNative.TurboModuleRegistry.getEnforcing('NitroModules');
+  turboModule = isTurboModuleEnabled ? _reactNative.TurboModuleRegistry.get('NitroModules'): _reactNative.NativeModules.NitroModules;
 } catch (e) {
   throw new _ModuleNotFoundError.ModuleNotFoundError(e);
 }
diff --git a/lib/module/turbomodule/NativeNitroModules.js b/lib/module/turbomodule/NativeNitroModules.js
index 826cd3edeceb9c7d040f4851c285ee5bb954bae4..664efb1c9dc60d405a30739aadf24f33b3f3d0f4 100644
--- a/lib/module/turbomodule/NativeNitroModules.js
+++ b/lib/module/turbomodule/NativeNitroModules.js
@@ -1,11 +1,12 @@
 "use strict";

-import { TurboModuleRegistry } from 'react-native';
+import { TurboModuleRegistry,NativeModules } from 'react-native';
 import { ModuleNotFoundError } from '../ModuleNotFoundError';
+const isTurboModuleEnabled = global.__turboModuleProxy != null;
 // 1. Get (and initialize) the TurboModule
 let turboModule;
 try {
-  turboModule = TurboModuleRegistry.getEnforcing('NitroModules');
+  turboModule = isTurboModuleEnabled?TurboModuleRegistry.get('NitroModules'): NativeModules.NitroModules;
 } catch (e) {
   throw new ModuleNotFoundError(e);
 }
diff --git a/src/turbomodule/NativeNitroModules.ts b/src/turbomodule/NativeNitroModules.ts
index a5e1a1bc53bf399be87b977767a18aafcb6ff251..92bf6f4bf2a8db010e5ec46e16c1079a31c45dfb 100644
--- a/src/turbomodule/NativeNitroModules.ts
+++ b/src/turbomodule/NativeNitroModules.ts
@@ -1,8 +1,8 @@
-import type { TurboModule } from 'react-native'
+import  {type TurboModule,NativeModules } from 'react-native'
 import { TurboModuleRegistry } from 'react-native'
 import { ModuleNotFoundError } from '../ModuleNotFoundError'
 import type { NitroModulesProxy } from '../NitroModulesProxy'
-
+const isTurboModuleEnabled = global.__turboModuleProxy != null;
 interface Spec extends TurboModule {
   /**
    * Installs `NitroModulesProxy` into this JS Runtime's `global`.
@@ -15,7 +15,7 @@ interface Spec extends TurboModule {
 // 1. Get (and initialize) the TurboModule
 let turboModule: Spec | undefined
 try {
-  turboModule = TurboModuleRegistry.getEnforcing<Spec>('NitroModules')
+  turboModule = isTurboModuleEnabled?TurboModuleRegistry.get<Spec>('NitroModules'): NativeModules.NitroModules
 } catch (e) {
   throw new ModuleNotFoundError(e)
 }
